Topic 0/src/point2/Demo.java

```diff
@@ -0,0 +1,26 @@
+package point2;
+
+import java.util.ArrayList;
+
+
+public class Demo {
+
+	public static void main(String[] args) {
+		
+		ArrayList<Room> aux= new ArrayList<Room>();
+		Builder builder = new LivingRoom();
+		Architect architecte = new Architect(builder);
+		architecte.build();
+		Room room = architecte.getRoom();
+		aux.add(room);
+		builder= new Kitchen();
+		architecte.setRoombuilder(builder);
+		architecte.build();
+		room= architecte.getRoom();
+		aux.add(room);
+		architecte.setRooms(aux);
```

Comment:
[Low] Instead of setting the whole list at the end, you could add a method in *Architect* called <code>addRoom</code> and you avoid this auxiliar variable.